From 1f75fa4b9c8b138585f13db094f3642f228c0ece Mon Sep 17 00:00:00 2001
From: Leorize <leorize+oss@disroot.org>
Date: Mon, 7 Jan 2019 08:42:27 +0000
Subject: os.execShellCmd: use WEXITSTATUS to retrieve exit code

According to POSIX, system() shall returns the termination status in the
format specified by waitpid(), which means WEXITSTATUS should be used to
retrieve the exit code portably.

This fixes execShellCmd on Haiku.

diff --git a/lib/pure/os.nim b/lib/pure/os.nim
index 96833056a0..a218121ed1 100644
--- a/lib/pure/os.nim
+++ b/lib/pure/os.nim
@@ -1298,7 +1298,7 @@ proc execShellCmd*(command: string): int {.rtl, extern: "nos$1",
   ## shell involved, use the `execProcess` proc of the `osproc`
   ## module.
   when defined(posix):
-    result = c_system(command) shr 8
+    result = WEXITSTATUS(c_system(command))
   else:
     result = c_system(command)
 
